esp/services/ws_topology/ws_topologyService.cpp

```diff
@@ -236,6 +236,16 @@ bool CWsTopologyEx::onTpLogFile(IEspContext &context,IEspTpLogFileRequest  &req,
     return true;
 }
 
+void CWsTopologyEx::validateFilePath(const char *file, const char *fileType, const char *compType, const char *compName)
+{
+    StringBuffer actualPath;
+    splitUNCFilename(file, nullptr, &actualPath, nullptr, nullptr);
+    if (containsRelPaths(actualPath)) //Detect a path like: /home/lexis/runtime/var/log/HPCCSystems/myesp/../../../
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid file path %s", actualPath.str());
+    if (!validateConfigurationDirectory(nullptr, fileType, compType, compName, actualPath))
```

Comment:
This is only going to work for log files if there isn't an override for a particular instance in the environment file, and then also only works because [INSTANCE] is the trailing component.  It probably works for most existing environment files, but not for all.

Reply:
Do we know where those special log file could be located? Should we add some settings to specify the paths to be allowed for retrieving the log files (and force users to modify their environment.xml)?

Reply:
They could theoretically be located anywhere.  I don't know if any real-life environments actually hit this.

Reply:
As I know, the function here is used for accessing the configuration files and log files of HPCC components (most likely through ECLWatch). Any suggestion about how to do with those special log files?